fix(isolated-declarations): optional parameter property misses `undefined` type

Dunqing opened the pull request:
[TypeScript Playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAYwDYEMDOa4AUoEsBuKMwAwhAHZoxQCuCM0A6rjABYDyYMuFKSWKKCgC2wYlEwBvAFABIMHkLFEFKrXrQAFHNlRgARxq49AEwBccNbnIBzADQ6IXHuT4B+C+RrCARsCgOsvI0Pki4CHBgIWEInNy8SB5wPhAQSMAo5IHyikTAkbnEcS7uFla22XooJhRIAJ5wVTXk9cUJSV6+-oEAlHCSAL7SA0A)

Input: 
```ts
export class PrivateConstructorWithOptionalParameters {
	private constructor(
		required: string,
		optional?: number,
		public publicOptional?: boolean,
		private privateOptional?: string,
		readonly readonlyOptional?: number,
	) {}
}
```

Output:

```ts
export declare class PrivateConstructorWithOptionalParameters {
    publicOptional?: boolean | undefined;
    private privateOptional?;
    readonly readonlyOptional?: number | undefined;
    private constructor();
}
```

optional parameter property should union the original type with `undefined`

Boshen commented:
### Merge activity

* **Jul 28, 1:22 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(isolated-declarations): optional parameter property misses `undefined` type (#12579)

[TypeScript Playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAYwDYEMDOa4AUoEsBuKMwAwhAHZoxQCuCM0A6rjABYDyYMuFKSWKKCgC2wYlEwBvAFABIMHkLFEFKrXrQAFHNlRgARxq49AEwBccNbnIBzADQ6IXHuT4B+C+RrCARsCgOsvI0Pki4CHBgIWEInNy8SB5wPhAQSMAo5IHyikTAkbnEcS7uFla22XooJhRIAJ5wVTXk9cUJSV6+-oEAlHCSAL7SA0A)

Input:
```ts
export class PrivateConstructorWithOptionalParameters {
	private constructor(
		required: string,
		optional?: number,
		public publicOptional?: boolean,
		private privateOptional?: string,
		readonly readonlyOptional?: number,
	) {}
}
```

Output:

```ts
export declare class PrivateConstructorWithOptionalParameters {
    publicOptional?: boolean | undefined;
    private privateOptional?;
    readonly readonlyOptional?: number | undefined;
    private constructor();
}
```

optional parameter property should union the original type with `undefined`
